src/main/java/io/supertokens/session/accessToken/AccessTokenSigningKey.java

```diff
@@ -90,29 +96,80 @@ synchronized void removeKeyFromMemoryIfItHasNotChanged(List<KeyInfo> oldKeyInfo)
         }
     }
 
-    public synchronized List<KeyInfo> getKey()
+    public synchronized void transferLegacyKeyToNewTable()
             throws StorageQueryException, StorageTransactionLogicException {
+        Storage storage = StorageLayer.getSessionStorage(main);
+
+        if (storage.getType() == STORAGE_TYPE.SQL) {
+            SessionSQLStorage sqlStorage = (SessionSQLStorage) storage;
+
+            // start transaction
+            sqlStorage.startTransaction(con -> {
+                KeyValueInfo legacyKey = sqlStorage.getLegacyAccessTokenSigningKey_Transaction(con);
+
+                if (legacyKey != null) {
+                    sqlStorage.addAccessTokenSigningKey_Transaction(con, legacyKey);
+                    sqlStorage.removeLegacyAccessTokenSigningKey_Transaction(con);
+                }
+                return legacyKey;
+            });
+        } else {
+            while (true) {
```

Comment:
Why do we need to put this in a while true loop?

Reply:
No need, it was part of a transaction with retry originally. I'll remove it.